traffic_ops/app/bin/checks/ToDSCPCheck.pl

```diff
@@ -105,179 +111,231 @@
+   #DEBUG "Debug matchList after push: " . Dumper($ds_info{ $ds->{id} }{matchList});
+}
+
+my $jdataserver = ();
+if (defined($args{s})) {
+   # TODO check the return on this
+   DEBUG "DEBUG Getting single server";
+   # This is a reference to a hash
+   $jdataserver = $ext->get( '/api/1.1/servers/hostname/' . $args{s} . '/details' );
+
+   # create an array
+   my @tmp = ();
+   # put the hash reference in the array
+   $tmp[0] = $jdataserver;
+   # clear $jdataserver
+   $jdataserver = ();
+   # create a reference to the array
+   $jdataserver = \@tmp;
+
+} else {
+   # This is a reference to an array
+   $jdataserver = $ext->get( Extensions::Helper::SERVERLIST_PATH );
+   unless (ref($jdataserver)) {
+      print "jdataserver is not a reference at all.\n";
+   }
+   if (ref($jdataserver) eq "SCALAR") {
+      print "jdataserver is a reference to a scalar.\n";
+   }
+   if (ref($jdataserver) eq "ARRAY") {
+      print "jdataserver is a reference to a array.\n";
+   }
+   if (ref($jdataserver) eq "HASH") {
+      print "jdataserver is a reference to a hash.\n";
+   }
+}
+# exit ();
+my %cdns        = ();
+my $jcdns       = $ext->get( '/api/1.1/cdns' );
+# convert the cdns to a hash
+foreach my $cdn ( @{$jcdns} ) {
+   DEBUG "DEBUG cdn_name: " . $cdn->{name};
+   $cdns{ $cdn->{name} } = $cdn;
+   DEBUG "DEBUG domain_name: " . $cdns{$cdn->{name}}->{domainName};
 }
 
-my %domain_name_for_profile = ();
-my $jdataserver = $ext->get( Extensions::Helper::SERVERLIST_PATH );
 foreach my $server ( @{$jdataserver} ) {
    next unless $server->{type} =~ m/^EDGE/;    # We know this is DSCP, so we know we want edges only
-   my $ip         = trim($server->{ipAddress});
-   my $ip6        = trim($server->{ip6Address});
-   my $host_name  = trim($server->{hostName});
-   my $fqdn       = $host_name.".".trim($server->{domainName});
-   my $interface  = trim($server->{interfaceName});
-   my $http_port  = $server->{tcpPort};
-   my $https_port = $server->{httpsPort};
-   my $status     = $server->{status};
-   my $details    = $ext->get( '/api/1.1/servers/hostname/' . $host_name . '/details.json' );
-   my $successful = 1; # assume all is good
-   $ip6 =~ s/\/\d+$//;
-
-   TRACE "Checking: ".$host_name;
+   if ( $server->{status} ne 'REPORTED' ) {
+      INFO "INFO Skipping server: $server->{hostName} because status is: $server->{status}";
+      next;
+   }
+   my @ip_addrs    = ( $server->{ipAddress}, $server->{ip6Address});
+   my $host_name   = trim($server->{hostName});
+   my $fqdn        = $host_name.".".trim($server->{domainName});
+   my $interface   = trim($server->{interfaceName});
+   my $status      = $server->{status};
+   my $port        = $server->{tcpPort}; # this is for both http and https
+   my $protocol    = "http";
+   my $ip_protocol = "ipv4";
+   my $domain_name = undef;
+   my $details     = $ext->get( '/api/1.1/servers/hostname/' . $host_name . '/details.json' );
```

Comment:
Same as above RE endpoints, but additionally avoid `.json` suffixes whenever possible - or at the very least consistently use or don't use `.json`.

Reply:
Removed